product_pricelist_margins/models/product_pricelist_item.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools.float_utils import float_is_zero
+
+
+class ProductPricelistItem(models.Model):
+    _inherit = "product.pricelist.item"
+
+    cost = fields.Float(
+        related="product_tmpl_id.standard_price",
+        digits="Product Price",
+    )
+    margin = fields.Float(
+        compute="_compute_margin",
+        digits="Product Price",
+    )
+    margin_percent = fields.Float(
+        string="Margin (%)",
+        compute="_compute_margin",
+    )
+
+    @api.depends("fixed_price", "product_tmpl_id", "cost")
+    def _compute_margin(self):
+        for item in self:
```

Comment:
```suggestion
        for item in self.filtered(lambda x: x.applied_on in ("1_product", "0_product_variant"):
```

The margin doesn'nt make sense for "product category" and "All products", right ?
(If yes, also hide the fields in all views if applied on is not correct).

Reply:
We need to set the default value to 0 for both the margin and margin_percent fields. Currently, when we click 'add a line' and have not selected a product yet, Odoo raises an error because margin and margin_percent are null. Since both fields are computed fields, we cannot declare a default value for them directly.

![image](https://github.com/OCA/product-attribute/assets/32189890/68035feb-8e43-417f-844c-ab7159b62b3f)

Reply:
you can then add : 

```
for item in self.filtered(lambda x: x.applied_on not in ("1_product", "0_product_variant"):
    item.margin = 0
    item.margin_percent = 0
```